packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -29,4 +29,35 @@ export class GOVUKFrontendComponent {
       throw new SupportError()
     }
   }
+
+  /**
+   * @protected
+   * @param {Schema} schema
+   * @param {Config[ConfigKey]} config
+   * @returns {boolean} Whether the config is valid
+   */
+  validateConfig(schema, config) {
+    if (!schema.anyOf) {
+      return true
+    }
+
+    return schema.anyOf.some(({ required }) =>
+      required.every((prop) => prop in config && config[prop])
+    )
+  }
 }
+
+/**
+ * @typedef {object} Schema
+ * @property {SchemaItem[]} [anyOf] - An array of conditions
+ */
+
+/**
+ * @typedef {object} SchemaItem
+ * @property {string[]} [required] - Whether the current item is required
```

Comment:
Think we best make `required` mandatory

Otherwise we'll potentially crash on `required.every()` above

```suggestion
 * @property {string[]} required - Whether the current item is required
```